baremetalpxeserver/baremetalpxeserver/pxeserveragent.py

```
@@ -557,11 +576,23 @@ def _create_preconfiguration_file(self, cmd):
             rendered_content = self._render_preseed_template(cmd, pxeserver_dhcp_nic_ip)
         elif cmd.preconfigurationType == 'autoyast':
             rendered_content = self._render_autoyast_template(cmd, pxeserver_dhcp_nic_ip)
+        elif cmd.preconfigurationType == 'autoinstall':
+            rendered_content = self._render_autoinstall_template(cmd, pxeserver_dhcp_nic_ip)
         else:
             raise PxeServerError("unkown preconfiguration type %s" % cmd.preconfigurationType)
 
         ks_cfg_name = cmd.pxeNicMac
         ks_cfg_file = os.path.join(self.KS_CFG_PATH, ks_cfg_name)
+        if cmd.preconfigurationType == 'autoinstall':
+            bash_r("mkdir -p %s" % ks_cfg_file)
```

Ruff (0.11.9)

`bash_r` may be undefined, or defined from star imports

(F405)